staging/src/k8s.io/apiserver/pkg/cel/environment/environment.go

```diff
@@ -175,7 +175,15 @@ type VersionedOptions struct {
 	//
 	// Optional.
 	RemovedVersion *version.Version
-
+	// FeatureEnabled returns true if these options are enabled by feature gates,
+	// and returns false if these options are not enabled due to feature gates.
+	//
+	// This takes priority over IntroducedVersion / RemovedVersion for the NewExpressions environment.
```

Comment:
takes priority in both directions?  If IntroducedVersion is after the desired compatible version, if the featuregate is enabled, the VersionedOption is enabled?

Reply:
Correct. This lets someone opt into an alpha feature in its first release and use a CEL feature unknown to the previous release, just like opting into an alpha feature will let you use API fields unknown to the previous release.